async-nats/src/jetstream/consumer/pull.rs

```diff
@@ -246,7 +252,110 @@ impl<'a> futures::Stream for Sequence<'a> {
     }
 }
 
-pub type Stream<'a> = stream::TryFlatten<Sequence<'a>>;
+pub struct Stream<'a> {
+    pending_messages: usize,
+    subscriber: Subscriber,
+    context: Context,
+    inbox: String,
+    subject: String,
+    batch_config: BatchConfig,
+    request: Option<BoxFuture<'a, Result<(), Error>>>,
+}
+
+impl<'a> Stream<'a> {
+    async fn stream(
```

Comment:
Maybe just inline this? because you're just forwarding Consumer to do simple subject and subscription construction. Overall this is more code and need to jump up and down to grok it.

Reply:
When builder comes in with next PR, this approach will be better. I would not inline it only to re-decouple in few hours :)